### PR TITLE
feat(replacements): add eslint-plugin-eslint-comments replacement

### PR DESCRIPTION
**Pull request by Jayllyz**
## Changes

`eslint-plugin-eslint-comments` is no longer maintained.

`@eslint-community/eslint-plugin-eslint-comments` is a direct fork which is actively maintained. It has new features, bugfixes and updated dependencies.

from: https://github.com/es-tooling/module-replacements/blob/main/docs/modules/eslint-plugin-eslint-comments.md

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [X] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/data/replacements.json [hunk 1]
```diff
@@ -211,6 +212,17 @@
...
   },
+  "eslint-plugin-eslint-comments-to-maintained-fork": {
+    "description": "Replace `eslint-plugin-eslint-comments` with a maintained fork: `@eslint-community/eslint-plugin-eslint-comments`.",
+    "packageRules": [
+      {
+        "matchDatasources": ["npm"],
+        "matchPackageNames": ["eslint-plugin-eslint-comments"],
+        "replacementName": "@eslint-community/eslint-plugin-eslint-comments",
+        "replacementVersion": "4.4.0"
+      }
```
Comment (rarkins) [inline lib/data/replacements.json:223]: Can we have a matchCurrentVersion=^4.0.0 ?

File: lib/data/replacements.json [hunk 2]
```diff
@@ -211,6 +212,17 @@
...
   },
+  "eslint-plugin-eslint-comments-to-maintained-fork": {
+    "description": "Replace `eslint-plugin-eslint-comments` with a maintained fork: `@eslint-community/eslint-plugin-eslint-comments`.",
+    "packageRules": [
+      {
+        "matchDatasources": ["npm"],
+        "matchPackageNames": ["eslint-plugin-eslint-comments"],
+        "replacementName": "@eslint-community/eslint-plugin-eslint-comments",
+        "replacementVersion": "^4.4.0"
+      }
```
Comment (rarkins) [inline lib/data/replacements.json:223]:
```suggestion
      {
        "matchDatasources": ["npm"],
        "matchPackageNames": ["eslint-plugin-eslint-comments"],
        "matchCurrentVersion": "^4.0.0",
        "replacementName": "@eslint-community/eslint-plugin-eslint-comments",
        "replacementVersion": "4.4.0"
      }
```

File: lib/data/replacements.json [hunk 3]
```diff
@@ -211,6 +212,18 @@
...
   },
+  "eslint-plugin-eslint-comments-to-maintained-fork": {
+    "description": "Replace `eslint-plugin-eslint-comments` with a maintained fork: `@eslint-community/eslint-plugin-eslint-comments`.",
+    "packageRules": [
+      {
+        "matchDatasources": ["npm"],
+        "matchPackageNames": ["eslint-plugin-eslint-comments"],
+        "matchCurrentVersion": "^4.0.0",
+        "replacementName": "@eslint-community/eslint-plugin-eslint-comments",
+        "replacementVersion": "4.4.0"
```
Comment (viceice) [inline lib/data/replacements.json:223]: is that the first new release? use the lowest new version

File: lib/data/replacements.json [hunk 4]
```diff
@@ -211,6 +212,18 @@
       }
     ]
   },
+  "eslint-plugin-eslint-comments-to-maintained-fork": {
+    "description": "Replace `eslint-plugin-eslint-comments` with a maintained fork: `@eslint-community/eslint-plugin-eslint-comments`.",
+    "packageRules": [
+      {
+        "matchDatasources": ["npm"],
+        "matchPackageNames": ["eslint-plugin-eslint-comments"],
+        "matchCurrentVersion": "^4.0.0",
```
Comment (rarkins) [inline lib/data/replacements.json:221]:
```suggestion
        "matchCurrentVersion": "^3.2.0",
```

Last version on https://www.npmjs.com/package/eslint-plugin-eslint-comments

File: lib/data/replacements.json (hunk 3 above)
Comment (rarkins) [inline lib/data/replacements.json:223]:
```suggestion
        "replacementVersion": "3.2.0"
```

This exists per https://www.npmjs.com/package/@eslint-community/eslint-plugin-eslint-comments/v/3.2.0

Comment (rarkins) [reply lib/data/replacements.json:223]: Good catch

**Review by viceice — APPROVED** [review]

**Pull request merged**